Add use cases in developer guide

xhphoong opened the pull request:
Add "mark", "delete", "edit", "add tag", "delete tag", "add date and time", "delete date and time", "add a patient to nurse", "deletes a patient from nurse" use cases.

Commit:
Add "mark" use case and modify "delete" use case in developer guide

Commit:
Modify "delete" and "mark" use case and add other use cases

Let's add "edit", "add tag", "delete tag", "add date and time",
"delete date and time", "add a patient to nurse",
"delete a patient from nurse" use cases.

xhphoong commented:
Pls check. If anything weird or missing, update here. Thankss

ou4jairmeiogusrjejiepr commented on docs/DeveloperGuide.md:
```diff
@@ -283,32 +283,422 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+      Use case restart. 
+
+* 1b. Medical administrator get a specific patient/nurse.
```
I am wondering if it is possible to omit this, because it seems like the number of steps is still the same, the only difference is that for 1b1, healthcare xpress displays only 1 record whereas for step 2, healthcare xpress displays >1 record. Since there is no difference in the number of steps, I don't think we need to handle this case differently. What are your thoughts?

xhphoong replied:
Hmm I also not sure about it cuz like I wrote that to show that not only list but can also use get n delete that specific person. But it is repetitive , so hmmm I also not very sure. 

Btw, one question, the textbook didnt say anything ah but can like add extensions for extensions (like the method is in extensions already but if do that method, it can lead to another extensions sth like that)

ou4jairmeiogusrjejiepr replied:
> Hmm I also not sure about it cuz like I wrote that to show that not only list but can also use get n delete that specific person. But it is repetitive , so hmmm I also not very sure.

Okay maybe we leave it like that for now.

> Btw, one question, the textbook didnt say anything ah but can like add extensions for extensions (like the method is in extensions already but if do that method, it can lead to another extensions sth like that)

Yeah should be can but try not to recurse too much haha

ou4jairmeiogusrjejiepr commented on docs/DeveloperGuide.md:
```diff
@@ -283,32 +283,422 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
       Use case resumes at step 2.
 
-*{More to be added}*
+* a. At any time, Medical administrator choose to exit the program.
```
Not sure about this, but should "a" here be "*a" to denote it can happen at any step?

xhphoong replied:
oo yes thanks for catching that.

ou4jairmeiogusrjejiepr commented on docs/DeveloperGuide.md:
```diff
@@ -283,32 +283,422 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+* 1b. Medical Administrator request to list nurses only.
+
+    Use case ends. 
```
Should an error message be shown here? Since the intended use of the function is to mark a patient, but the user inputs a nurse instead.

xhphoong replied:
Okkie noted, will change it.

ou4jairmeiogusrjejiepr commented on docs/DeveloperGuide.md:
```diff
@@ -283,32 +283,422 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+    * 1c2. Medical administrator requests to mark that patient.
+
+      Use case resumes from step 4.   
```
As mentioned above I'm not sure if this can be omitted. Let me know what you think!

ou4jairmeiogusrjejiepr commented on docs/DeveloperGuide.md:
```diff
@@ -283,32 +283,422 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+  Use case ends.
+
+**Use case: UC06 - Add a date and time to patient**
```
Just a small suggestion but maybe we can name this "home visit date and time" instead of just "date and time"

xhphoong replied:
Okay will change

ou4jairmeiogusrjejiepr reviewed (commented):
LGTM! Just a few minor suggestions, let me know what you think! If you find it's not suitable that's fine too

Commit:
Fix the format, make minor changes to some use cases

Let's ,
- change all a to *a at the extensions sections
- modify the extension for mark, add date and time, delete date and
  time
- modify the use case description for remove patient from nurse, add
  date and time and delete date and time

Commit:
Modify existing use cases and Add new use cases

Commit:
Fix minor issue

ou4jairmeiogusrjejiepr reviewed (approved):
LGTM, thank you!
